packages/report-server/src/__tests__/reportBuilder/query/QueryBuilder.test.ts

```diff
@@ -82,23 +82,31 @@ describe('QueryBuilder', () => {
   const outputQuery = (input: Partial<FetchReportQuery> = {}) => ({
     hierarchy: HIERARCHY,
     organisationUnitCodes: ['TO'],
-    period: getPeriodString([2017, 1], [2020, 12]),
+    period: getPeriodString([2017, 1], [2020, 12, 15]),
     startDate: '2017-01-01',
-    endDate: '2020-12-31',
+    endDate: '2020-12-15',
     ...input,
   });
 
   const getPeriodString = (
-    [startYear, startMonth]: [number, number],
-    [endYear, endMonth]: [number, number],
+    [startYear, startMonth, startDay = 1]: [number, number, number?],
+    [endYear, endMonth, endDay = 1]: [number, number, number?],
   ) => {
     const periods = [];
 
     for (let year = startYear; year <= endYear; year++) {
       const currentStartMonth = year === startYear ? startMonth : 1;
       const currentEndMonth = year === endYear ? endMonth : 12;
       for (let month = currentStartMonth; month <= currentEndMonth; month++) {
-        periods.push(`${year}${month.toString().padStart(2, '0')}`);
+        if (month === currentEndMonth && year === endYear && endDay > 1) {
+          for (let day = startDay; day <= endDay; day++) {
+            periods.push(
+              `${year}${month.toString().padStart(2, '0')}${day.toString().padStart(2, '0')}`,
+            );
+          }
+        } else {
+          periods.push(`${year}${month.toString().padStart(2, '0')}`);
+        }
```

Comment:
I think maybe the logic here is not _quite_ right. What would happen if someone called: 
`getPeriodString([2020, 11, 13], [2020, 12, 14])`?

Happy to pair with ya on this if you want, though I'll give a bit of a hint and suggest that you may need to use [this](https://stackoverflow.com/questions/1184334/get-number-days-in-a-specified-month-using-javascript)